Sources/NIOSSL/SSLContext.swift

```diff
@@ -472,6 +485,32 @@ extension NIOSSLContext {
         if result == 0 {
             let errorStack = BoringSSLError.buildErrorStack()
             throw BoringSSLError.unknownError(errorStack)
+        } else if sendCANames, !isDirectory {
+            // For single CA file, add the CA name from the trust root.
+            // This could be from a location like /etc/ssl/cert.pem as an example.
+            CNIOBoringSSL_SSL_CTX_set_client_CA_list(context, CNIOBoringSSL_SSL_load_client_CA_file(path))
+        } else if sendCANames, isDirectory {
+            // If the path that is passed in is a directory, scan the directory and gather up the PEM files.
+            var pemFilePaths: [String] = []
+            if let dir = opendir(path) {
+                while let dirent: UnsafeMutablePointer<dirent> = readdir(dir) {
+                    let fileName = withUnsafePointer(to: &dirent.pointee.d_name) { (ptr) -> String in
+                        return ptr.withMemoryRebound(to: CChar.self, capacity: Int(ptr.pointee.0) + 1) { String(cString: $0) }
+                    }
+                    if fileName.suffix(4) == ".pem" {
+                        pemFilePaths.append(path + fileName)
+                    }
+                }
+                closedir(dir)
+            }
+            // Load the PEM files one by one and perform the following functions:
+            // 1. addRootCertificate to the context.
+            // 2. addCACertificateNameToList to send the CA names during the handshake.
+            for path in pemFilePaths {
+                let cert = try NIOSSLCertificate(file: path, format: .pem)
+                try NIOSSLContext.addRootCertificate(cert, context: context)
```

Comment:
I don't think this call is needed: we've already added the certificate in the load verify locations call.

Reply:
It does look like `SSL_CTX_load_verify_locations` is supposed to load the certificate.  Now, with and without the usage of `NIOSSLContext.addRootCertificate` the handshake succeeds from an XCTest.  However, when removing `NIOSSLContext.addRootCertificate`, and WITH the certificate chain is installed on the machine, `BoringSSLError` throws  `TLSV1_ALERT_UNKNOWN_CA` which does not sit right to me, especially because the other two routes for `testFullVerificationWithCANamesFromCertificate` and `testFullVerificationWithCANamesFromFile` do not end up this way.  To work around this I shimmed in `clientConfig.trustRoots = .certificates([])` only in the test case for `testFullVerificationWithCANamesFromDirectory`.  @Lukasa if you have any other thoughts here on this matter, let me know.

Reply:
Sorry, can you clarify the context where we get an unexpected failure?

Reply:
@Lukasa sure, the failure existed in the test case for `testFullVerificationWithCANamesFromDirectory`.
If a machine running the test case has the full certificate chain installed on the machine, then the test case will receive a failure for `TLSV1_ALERT_UNKNOWN_CA`.  Now, I suspect that this is only the case for self signed certificates, as that is what I am using here. The strange part about this is that I didn't see this failure with any of the other test cases that test for CA names.  I fixed the failure in the test case, but just wanted to see if there was anything that stuck out to you here.